google-cloud-firestore/src/main/java/com/google/cloud/firestore/ExponentialBackoff.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.firestore;
+
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+/** Helper to implement exponential backoff. */
+class ExponentialBackoff {
```

Comment:
It's slightly annoying that we have yet another exponential backoff implementation here. GAX already has one (https://github.com/googleapis/gax-java/blob/master/gax/src/main/java/com/google/api/gax/retrying/ExponentialRetryAlgorithm.java). Maybe it's ok since it is package-private; maybe add a note in the documentation stating that this shouldn't be exposed in the future as public due to concerns about duplicating surface?

Reply:
I'm not opposed to migrating to this implementation. Let me take a look at how involved this is.